include/glow/Support/CompilationLog.h

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef GLOW_SUPPORT_COMPILATIONLOG_H
+#define GLOW_SUPPORT_COMPILATIONLOG_H
+
+#include "llvm/ADT/StringRef.h"
+
+#include <string>
+
+namespace glow {
+
+/// A class that defines the compilation scope.
+///
+/// Example:
+class CompilationScope {
+private:
+  /// Name of current scope.
+  /// Example: "ExecutionEngine::compile"
+  std::string scopeName_;
+  // Full scope name that reprensents the entire compilation phases till now.
+  std::string fullScopeName_;
```

Comment:
Is this used anywhere?

Reply:
Probably will be needed in the future. I'll remove it right now.